### PR TITLE
Fds 134 sergey. Parse and validate tokens

### PR DESCRIPTION
**Pull request by Baraulia**

### CODE REVIEW

**Comment by Ann40a** [comment]
Please update the title of the PR to say about made changes rather than your name ;)

File: GRPC/grpcServer/grpcserver.go [hunk 1]
```diff
@@ -33,42 +32,33 @@ func NewGRPCServer(service *service.Service) {
...
+func (g *GRPCServer) TokenGenerationByUserId(ctx context.Context, user *authProto.User) (*authProto.GeneratedTokens, error) {
 	if user.RoleId == 0 {
-		role, err := g.service.GetRoleById(int(user.UserId))
+		roleId, err := g.service.GetRoleByUserId(int(user.UserId))
 		if err != nil {
 			logger.Errorf("GetRoleById:%s", err)
 			return nil, fmt.Errorf("GetRoleById:%w", err)
 		}
-		user.RoleId = int32(role.ID)
+		user.RoleId = int32(roleId)
```
Comment (Ann40a) [inline GRPC/grpcServer/grpcserver.go:57]: if role should be integer why service returns it as string. If it is string, why you try to convert it to integer?

File: cmd/main.go [hunk 2]
```diff
@@ -17,6 +16,7 @@ import (
 
 // @title Authorization Service
 // @description Authorization Service for Food Delivery Application
+
```
Comment (Ann40a) [inline cmd/main.go]: is it needed? I mean empty line

File: handler/router.go [hunk 3]
```diff
@@ -21,6 +21,10 @@ func NewHandler(services *service.Service, logger logging.Logger) *Handler {
 func (h *Handler) InitRoutes() *gin.Engine {
 	router := gin.New()
 	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
+	router.Use(
+		h.CorsMiddleware,
```
Comment (Ann40a) [inline handler/router.go:25]: Do you use userIdentity middleware?

File: service/auth_service.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	return &authProto.UserRole{
+		UserId:      claims["UserId"].(int32),
+		Role:        role.Name,
+		Permissions: stringPerms,
+	}, nil
+}
+
+func (a *AuthService) RefreshTokens(refreshToken string) (*authProto.GeneratedTokens, error) {
+	parseToken, err := jwt.Parse(refreshToken, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
```
Comment (Ann40a) [inline service/auth_service.go]: this part is repeated between parsing of access and refresh. Can it be reused?

File: service/role_perm_service.go [hunk 5]
```diff
@@ -48,6 +52,10 @@ func (s *RolePermService) GetAllPerms() ([]model.Permission, error) {
 	return s.repo.RolePerm.GetAllPerms()
 }
 
-func (s *RolePermService) AddRoleToUser(user *authProto.User) error {
-	return s.repo.AddRoleToUser(user)
+func (s *RolePermService) AddRoleToUser(user *authProto.User) (*authProto.ResultBinding, error) {
+	err := s.repo.AddRoleToUser(user)
+	if err != nil {
+		return &authProto.ResultBinding{Result: false}, err
```
Comment (Ann40a) [inline service/role_perm_service.go]: why service converts the result to object from proto? Looks like controller should do it, not service

**Review by Ann40a — APPROVED** [review]

File: GRPC/grpcServer/grpcserver.go (hunk 1 above)
Comment (Baraulia) [reply GRPC/grpcServer/grpcserver.go:57]: Service returns roleId as int not as string, but GRPC functions do not know about such type. They accept int32 or int64.

File: cmd/main.go (hunk 2 above)
Comment (Baraulia) [reply cmd/main.go]: fixed

File: service/auth_service.go (hunk 4 above)
Comment (Baraulia) [reply service/auth_service.go]: Yes, it can. I fixed it.

File: service/role_perm_service.go (hunk 5 above)
Comment (Baraulia) [reply service/role_perm_service.go]: fixed

File: handler/router.go (hunk 3 above)
Comment (Baraulia) [reply handler/router.go:25]: No, it is not an aim of this task. I plan to use this middleware later

**Pull request merged**